### DISCUSSION TITLE
Reassess Supplier Defer Definition

### DISCUSSION THREAD

**Discussion by j---** [CERTCC/SSVC#260] [Ideas]
https://github.com/CERTCC/SSVC/blob/82363313b2f0c17308bce829753641bda77e73e3/doc/md_src_files/040_stakeholders-scope.md?plain=1#L135C7-L135C7

`| Defer              | Do not work on the patch at present. |`

I think we could give clearer guidance on suppliers actions here. Based on discussion with @thomas-proell and @bs37208.

For a supplier, there probably should be a definition of Defer about when the vulnerability will be remediated, rather than the rather over-simplified "do not work on patch at present".

What do we think about tying Supplier actions in to release types related to SemVer? As in, *defer* would mean "fix in the next minor version release or equivalent time line". 
Then *scheduled* would track to fixing in the next scheduled patch release version. Naturally, *out-of-cycle* would mean issuing an out-of-cycle patch. I think *immediate* also leads to an out of cycle patch, but with different advisory and notifications around it, and probably with different resources devoted to getting it done faster. I'm not imagining a substantive change to the three levels above defer, just shifting the language to be similar to that for defer. 

I think we still have to leave space for developers to define their own software release cadences, even though of course we still have the problem of stacking lots of 90-day patch release time lines in a supply chain might mean that patches don't fully propagate for years. I think we have to work that separately, though, as getting people on shorter scheduled maintenance windows -- as in, that's not in scope for SSVC to solve by itself.

**Comment by ahouseholder** [comment]
I don't think we should lash our boat to SemVer as there are enough viable alternatives and I don't want to make it harder for folks doing continuous deployment to use SSVC.

To the broader point about the meaning of _Defer_ though, I think there are a variety of possible meanings (and we would do well to figure out which of them we intend). A few examples:

1. _Defer_ = "prioritize and handle within the usual bug handling process", taking it out of the "vulnerability response" queue and placing it in the general bug fixing pipeline? 
2. _Defer_ = "wontfix unless additional information arrives to change our mind"
3. _Defer_ = "take no action now other than to reassess in _timeframe_"

I think we had the second one in mind originally, and the reason it was _Defer_ instead of _Close_ is that _all_ decisions to take no further action are reversible (by deciding to take further action).

3 is almost the same as 2, although it makes the revisiting explicit. However, it would need to be a stakeholder-specified timeframe to address their operational tempo. Perhaps even a case-specific timeframe based on environmental volatility. But that might lead to lead to the potential of a suite of deferral outcomes based on timeframes: _defer-1_, _defer-7_, _defer-30_, _defer-90_ and now you're in the ironic state of prioritizing your deprioritizations.

Taking things in a slightly different (but not totally off-topic) direction, many IT problem / change management practices (think ITIL or whatever the cool kids are doing these days) have specific SLEs attached to issue severity levels. In principle, an SSVC deployer organization could choose to map our generic decision outcomes directly to their Severity levels or SLEs, for example:
- _Defer_ = Sev-4 = 90 days
- _Scheduled_ = Sev-3 = 30 days
- _Out-of-Cycle_ = Sev-2 = 7 days
- _Immediate_ = Sev-1 = 1 day

I know less about how idiosyncratic (or not) real-world developer bugfix SLEs can be, but it seems plausible that there might be some existing model(s) to map to? Again, as with the deployer example above though, I think we'd have to just highlight in the doc that "when we say _Defer_, you should figure out how that maps onto _your_ process, and then decide if the tree you are using is scaled appropriately to _your work priority categories_".

**Reply by j---** [reply]
Thanks for writing this out. SemVer can one option for how it can be represented, and SLEs can be another way of representing it. I agree we don't need to tie ourselves to just one. 

I think the broader point is that we currently have it defined as something like your 2 or 3. And I think it should be more like what you have as 1.
